src/js/core/methods/helpers/cardanoCertificate.js

```diff
@@ -0,0 +1,163 @@
+/* @flow */
+
+import { validateParams } from './paramsValidator';
+import { CERTIFICATE_TYPE, POOL_RELAY_TYPE } from '../../../constants/cardano';
+import { validatePath } from '../../../utils/pathUtils';
+import { ERRORS } from '../../../constants';
+
+import type {
+    CardanoCertificate,
+    CardanoPoolParameters,
+    CardanoPoolMargin,
+    CardanoPoolOwner,
+    CardanoPoolRelay,
+    CardanoPoolMetadata
+} from '../../../types/networks/cardano';
+
+import type {
+    CardanoTxCertificate,
+    CardanoPoolParameters as CardanoPoolParametersMessage,
+    CardanoPoolOwner as CardanoPoolOwnerMessage,
+    CardanoPoolRelay as CardanoPoolRelayMessage,
+} from '../../../types/trezor/protobuf';
+
+const ipv4AddressToHex = (ipv4Address: string): string => {
+    return Buffer.from(ipv4Address.split(".").map((ipPart) => parseInt(ipPart))).toString('hex')
+}
+
+const ipv6AddressToHex = (ipv6Address: string): string => {
+    return ipv6Address.split(":").join("")
+}
+
+const validatePoolMargin = (margin: CardanoPoolMargin) => {
+    validateParams(margin, [
+        { name: 'numerator', type: 'string', obligatory: true },
+        { name: 'denominator', type: 'string', obligatory: true },
+    ]);
+}
+
+const validatePoolMetadata = (metadata: CardanoPoolMetadata) => {
+    validateParams(metadata, [
+        { name: 'url', type: 'string', obligatory: true },
+        { name: 'hash', type: 'string', obligatory: true },
+    ]);
+}
+
+const validatePoolRelay = (relay: CardanoPoolRelay) => {
+    validateParams(relay, [
+        { name: 'type', type: 'number', obligatory: true },
+    ]);
+
+    switch (relay.type) {
+        case POOL_RELAY_TYPE.SingleHostIp: {
+            validateParams(relay, [
+                { name: 'ipv4Address', type: 'string', obligatory: !relay.ipv6Address },
+                { name: 'ipv6Address', type: 'string', obligatory: !relay.ipv4Address },
+                { name: 'port', type: 'number', obligatory: true },
+            ]);
+        }
```

Comment:
Aren't there missing `break;`s here?

Reply:
yes, I had it already locally fixed but forgot to push, sorry